### PR TITLE
Add gates ionq

### PR DESCRIPTION
**Pull request by JamesB-1qbit**
Add controlled rotations, swap, and phase gates. Tested on IonQ noisy simulator.

### CODE REVIEW

**Review by AlexandreF-1qbit — APPROVED** [review]

File: tangelo/linq/translator/translate_json_ionq.py [hunk 1]
```diff
@@ -55,22 +65,23 @@ def translate_json_ionq(source_circuit):
     json_gates = []
     for gate in source_circuit._gates:
         if gate.name in {"H", "X", "Y", "Z", "S", "T"}:
-            json_gates.append({'gate': GATE_JSON_IONQ[gate.name], 'target': gate.target[0]})
+            json_gates.append({'gate': GATE_JSON_IONQ[gate.name], 'targets': gate.target})
```
Comment (ValentinS4t1qbit) [inline tangelo/linq/translator/translate_json_ionq.py:68]: I'm a bit confused here. I have a feeling we are only dealing with 4 different situations:

- A list of gates that take no control, no parameters (SWAP, X, Y, Z, H, S, T)
- A list of gates that take parameters but no controls (RX, RY, RZ, PHASE, XX)
- A list of gates that take controls but no parameters (CX, CY, CZ, CNOT)
- A list of gates that take both (CRX, CRY, CRZ, CPHASE)

Couldn't we refactor this to only have 4 branches, now that controls and targets keywords are used everywhere ?

File: tangelo/linq/tests/test_translator.py [hunk 2]
```diff
@@ -382,31 +382,16 @@ def test_json_ionq(self):
...
-                                   {'gate': 'x', 'target': 1},
-                                   {'gate': 'rx', 'rotation': 1.5707963267948966, 'target': 0},
-                                   {'gate': 'h', 'target': 2},
-                                   {'control': 0, 'gate': 'cnot', 'target': 1},
-                                   {'gate': 'rz', 'rotation': 12.566170614359173, 'target': 2}],
+        ref_circuit = {'circuit': [{'gate': 'x', 'targets': [0]},
+                                   {'gate': 'x', 'targets': [1]},
+                                   {'gate': 'rx', 'rotation': 1.5707963267948966, 'targets': [0]},
+                                   {'gate': 'h', 'targets': [2]},
+                                   {'controls': [0], 'gate': 'x', 'targets': [1]},
```
Comment (ValentinS4t1qbit) [inline tangelo/linq/tests/test_translator.py:389]: Could we respect order here and start with gate name, for readability ?

**Review by ValentinS4t1qbit — CHANGES_REQUESTED** [review]
I think we can simplify things a bit on the translator side.
Thanks for looking into this, this is definitely an improvement in both features and I believe simplicity as well :)

**Review by ValentinS4t1qbit — APPROVED** [review]
Thanks, looks good !

**Pull request merged**